core/kernel/tee_ta_manager.c

```diff
@@ -277,6 +278,58 @@ static TEE_Result check_client(struct tee_ta_session *s, const TEE_Identity *id)
 	return TEE_SUCCESS;
 }
 
+/*
+ * Check if invocation parameters matches TA properties
+ *
+ * @sess - current session handle
+ * @param - already identified memory references hold a valid 'mobj'.
+ *
+ * Policy:
+ * - All TAs can access 'non-secure' shared memory.
+ * - All TAs can access TEE private memory (seccpy)
+ * - Only SDP flaged TAs can accept SDP memory references.
+ */
+#ifndef CFG_WITH_SDP
+static bool check_params(struct tee_ta_session *sess __unused,
+			 struct tee_ta_param *param __unused)
+{
+	/*
+	 * When CFG_WITH_SDP is not enable, no SDP memory reference is allowed
+	 * at OP-TEE core entry. Hence here all TAs have same permissions
+	 * regarding memory reference parameters.
+	 */
+	return true;
+}
+#else
+static bool check_params(struct tee_ta_session *sess,
+			 struct tee_ta_param *param)
+{
+	int n;
+
+	/*
+	 * When CFG_WITH_SDP is enable, OP-TEE entry allows SHM and SDP memory
+	 * references. Only TAs flagged SDP can access SDP memory references.
+	 */
+	if (sess->ctx->flags & TA_FLAG_SECURE_DATA_PATH)
+		return true;
+
```

Comment:
This piece of code prevents a user TA from invoking a static TA with SDP memory reference parameters.
I think this is mandatory when the client is non secure. But i client is secure (another TA), shouldn't we allow SDP TAs to invoke a pseudo-TA with SDP memrefs ?

Proposal#1: forbid pseudo-TA to get SDP memrefs (this may prevent a SDP specific service to be implemented as a pseudo-TA).
Proposal#2: add `flags` property to be defined by the macro `static_ta_register()` and let such pseudo-TAs deal with client identity and SDP memrefs.
Proposal#3: check here if invoked TA is a userTA and client is secure or not.

I would tend toward promoting proposal #2.